lib/FusionInventory/Agent/Tools/Hardware.pm

```diff
@@ -540,18 +559,56 @@ sub _getMacAddress {
     # fallback on ports addresses (IF-MIB::ifPhysAddress) if unique
     my $addresses_oid = ".1.3.6.1.2.1.2.2.1.6";
     my $addresses = $snmp->walk($addresses_oid);
+
+    # filter keeping mac with a defined and associated ip otherwise takes all macs
+    my $ips = $snmp->walk('.1.3.6.1.2.1.4.20.1.2');
+    my @macs = ();
+    @macs = grep { defined } map { $addresses->{$_} } values %{$ips}
+        if (keys(%{$ips}));
+    @macs = grep { defined } values %{$addresses}
+        unless @macs;
```

Comment:
I still tested this way, but I found cases where the first grep generates an empty list and the following grep finds a finally good list. I also found case where  a merged list avoid to find a good mac address even if one exists.

I'll add comments to explain the need and avoid confusion.